### PR TITLE
(issue #64) add `always_list` parameter (fparse/fload)

### LINKED ISSUE #64: Add 'enforce_lapply' or similar to fparse/fload
It is a great feature of the package that one does not need to call `lapply()` on character vectors since `fparse\fload` are already vectorized. However, this also means that the output type depends on whether the input contains a single element or multiple elements, e.g.:

```
inp2 <- as.character(1:2)
RcppSimdJson::fparse(inp2)
# [[1]]
# [1] 1
# 
# [[2]]
# [1] 2

inp1 <- as.character(1)
RcppSimdJson::fparse(inp1)
# [1] 1
```

So to be on the safe side, one has to check if the input vector is of length 1, and if yes, one has to wrap the result in a list. E.g.:

```
safe_fparse <- function(json, ...) {
  out <- RcppSimdJson::fparse(json, ...)
  if (length(json) == 1L) out <- list(out)
  out
}
safe_fparse("1")
# [[1]]
# [1] 1
```

What about adding an argument to fparse/fload which **enforces** that each element of the input json is wrapped in a list?

### PR DESCRIPTION
**Pull request by knapply**
Resolves #64.

### CODE REVIEW

**Review by eddelbuettel — APPROVED** [review]
Looks good as usual!  I can deal with the codecov nag separately.

I think I would have put the new named argument at the end by some time honoured convention though ...

**Comment by knapply** [comment]
> Looks good as usual! I can deal with the codecov nag separately.
> 
> I think I would have put the new named argument at the end by some time honoured convention though ...

The codecov should be fixed momentarily.  The rationale behind the argument order is that the arguments that are applicable to both `fparse()`/`fload()` go before the ones that are only applicable to `fload()`, but I can change that if desired.

**Comment by eddelbuettel** [comment]
Makes sense, I didn't pick up on that.

File: R/fload.R [hunk 1]
```diff
@@ -196,7 +196,7 @@ fload <- function(json,
     )
 
     if (always_list && length(json) == 1L) {
-        `attr<-`(list(out), "names", names(json))
+        `names<-`(list(out), names(json))
```
Comment (tdeenes) [inline R/fload.R:199]: `setNames()` is the more idiomatic way to set names for an object.

**Comment by tdeenes** [comment]
Thank you very much for the open-mindedness and the quick implementation!

File: R/fload.R (hunk 1 above)
Comment (knapply) [reply R/fload.R:199]: I suppose, but it's the exact same behavior without needing to add `{stats}` to the `DESCRIPTION`'s `Imports:`.

Comment (eddelbuettel) [reply R/fload.R:199]: Maybe, maybe not. `setNames()` is from `stats`, reading help suggests it is "just a short form" for assigning and naming.  I am cool with `base:names()`.  Can't argue with `base` even if there are shortcuts :)

Comment (tdeenes) [reply R/fload.R:199]: Indeed, I forgot about the fact that `setNames` comes from **stats**. Sorry for the noise :)

**Pull request merged**